### ISSUE TITLE
[ERROR] Fast-Path error: Bitmap error: Invalid first row size, must be 0

### ISSUE THREAD

**Issue by dheater** [Devolutions/IronRDP#34]
# Reproduction
## Setup

Deploy a Windows server 2022 machine in OpenStack and verify that you can establish an RDP connection with a known good RDP client.

Build and run latest IronRDP client from the command line.

Connect to the server using the IronRDP client

## Run

Run IronRDP from the command line like this:

cargo run – 192.168.10.10:3389 --username username --password ********

The connection will terminate

[ERROR] Fast-Path error: Bitmap error: Invalid first row size, must be 0, but got: 17197

You should find an error similar to above at the end of `ironrdp_client.log`

**Comment by dheater** [comment]
Instead of panic, seems like we should handle this error in a way that allows the client to make a best effort to continue.

Is non-zero first row now valid? (update to spec?)

**Comment by awakecoding** [comment]
@dheater I agree we shouldn't panic an catch a protocol error to gracefully disconnect, but in this case, this is more likely an unrecoverable error. Make sure that the RDP server is configure for RDP 7.1 RemoteFX, otherwise you will end up getting RDP6 GDI messages, and that part of the specification is partially implemented in IronRDP. RDP 7/8 is actually a lot easier to get working than the old stuff, I would suggest you make progress without getting involved with the RDP6 GDI messages (yet).

**Comment by dheater** [comment]
>  Make sure that the RDP server is configure for RDP 7.1 RemoteFX

I haven't found where I can configure the server protocol version (MS native server).  Any tips?

**Comment by CBenoit** [comment]
Is the panic triggered at the IronRDP library level, or is it triggered by the IronRDP demonstration client? The latter is an issue as we don't want to library to cause the user application to abort, but the former is probably intentional (simply stop execution on unrecoverable error).

**Comment by dheater** [comment]
The error is bubbling up from the library. I dropped a PR with my workaround: https://github.com/Devolutions/IronRDP/pull/35

**Comment by awakecoding** [comment]
As reference, here are the similar code paths in FreeRDP. We should make sure to skip messages that we don't yet support, such that we can just throw a warning in the logs instead of dropping the connection:

https://github.com/FreeRDP/FreeRDP/blob/master/libfreerdp/core/fastpath.c#L367
https://github.com/FreeRDP/FreeRDP/blob/master/libfreerdp/core/fastpath.c#L264
https://github.com/FreeRDP/FreeRDP/blob/master/libfreerdp/core/update.c#L201
https://github.com/FreeRDP/FreeRDP/blob/master/libfreerdp/core/update.c#L86

The current level of support mostly uses RDP 7.1 RemoteFX, which uses the "surface commands"
